examples/http-server/main_test.go

```diff
@@ -26,15 +28,31 @@ func doGetRequest(t *testing.T, getPath string) int {
 	return resp.StatusCode
 }
 
+func doPostRequest(t *testing.T, postPath string, data []byte) (int, []byte) {
+	t.Helper()
+	resp, err := http.Post(postPath, "application/x-www-form-urlencoded", bytes.NewBuffer(data))
+	if err != nil {
+		log.Fatalln(err)
+	}
+	respBody, err := io.ReadAll(resp.Body)
+	resp.Body.Close()
+	if err != nil {
+		log.Fatalln(err)
+	}
+	return resp.StatusCode, respBody
+}
+
 func TestHttpServer(t *testing.T) {
 	tests := []struct {
-		name      string
-		path      string
-		expStatus int
-		envVars   map[string]string
+		name            string
+		path            string
+		expStatus       int
+		envVars         map[string]string
+		body            []byte // if body is populated, POST request is sent
+		expResponseBody []byte
```

Comment:
While I like adding tests, I am not sure this detailed tests should be here. I think the logic should be tested in the middleware itself and here we could have one to demonstrate how it works maybe?